controller/configentry_controller.go

```diff
@@ -167,9 +185,16 @@ func (r *ConfigEntryController) ReconcileEntry(
 		return r.syncFailed(ctx, logger, crdCtrl, configEntry, ConsulAgentError, err)
 	}
 
+	// Check if the config entry is managed by our datacenter.
+	// Do not process resource if the entry was not created within our datacenter
+	// as it was created in a different cluster within the federation.
+	if entry.GetMeta()[common.DatacenterKey] != r.DatacenterName {
+		return r.syncFailed(ctx, logger, crdCtrl, configEntry, ExternallyManagedConfigError, errors.New("config entry managed in different datacenter"))
```

Comment:
Should this error contain the other dc's name?

Reply:
I on the fence with this one. I guess it is ok to share this information? 

Also, there is a good chance it might not have that info either ie it could just be blank (it would require every "system") that creates a config entry to set the meta fields.

Reply:
Well, i added it, but we should test this a little bit